.assets/SetupFabric.md

```diff
@@ -0,0 +1,60 @@
+The service principal that you have created in step 1 needs to be added to the workspace. Go to the workspace and click on **Manage access** and search for your service principal. Select the service principal and click on **Add**.
+![Access Management](/.assets/manageAccessFabric.png)
+
+*it is possible that you cannot see the service principal then go to the admin tenant settings and enable the setting "Allow service principals to use Power BI API's"*
+![Fabric Tenant Settings](/.assets/fabricTenantSettings.png)
+
+## Configuring the Dynamics 365 Business Central
+Install the extension into BC using the code given in the [businessCentral](/businessCentral) folder using the general guidance for [developing extensions in Visual Studio code](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-dev-overview#developing-extensions-in-visual-studio-code). 
+
+The app exposes [3 permission sets](/businessCentral/permissions/) for different user roles that work with this app. Remember to assign the right permission set to the user, based on the scope of their tasks:
+1. `ADLSE - Setup`- The permission set to be used when administering the Azure Data Lake Storage export tool.
+1. `ADLSE - Execute`- The permission set to be used when running the Azure Data Lake Storage export tool.
+1. `ADLSE - Track Delete`- The permission set used to register the deletion of any record, so that the information of it being deleted can be conveyed to the Azure data lake. A user who deleting a table that is configured for export by `bc2adls` should have this permission.
+
+Once you have the `Azure Data Lake Storage Export` extension deployed, open the `Page 82560 - Export to Azure data lake Storage`. In order to export the data from inside BC to the data lake, you will need to add a configuration to make BC aware of the location in the data lake.
+
+### Step 5. Enter the BC settings
+Let us take a look at the settings show in the sample screenshot below,
+- **Storage Type** choose here the storage type. Choose "Microsoft Fabric"
+- **Tenant ID** The tenant id at which the app registration created above resides (refer to **b)** in the picture at [Step 1](/.assets/Setup.md#step-1-create-an-azure-service-principal))
+- **Workspace** The workspace in your Microsoft Fabric environment where the lakehouse is located. This can also be a GUID.
+- **Lakehouse** The name or GUID of the lakehouse inside the workspace.
+- **Max payload size (MiBs)** The size of the individual data payload that constitutes a single REST Api upload operation to the data lake. A bigger size will surely mean less number of uploads but might consume too much memory on the BC side. Note that each upload creates a new block within the blob in the data lake. The size of such blocks are constrained as described at [Put Block (REST API) - Azure Storage | Microsoft Docs](https://docs.microsoft.com/en-us/rest/api/storageservices/put-block#remarks).
```

Comment:
It is used in the code. It looks like also de Onelake API works with it.

Reply:
While using Azure Data Lake the API call is using the PUT method with and creates and Block. In case of Fabric, where using the Patch method and appending the file with the position marker. See the code blocks below (the second method for Fabric could use a better name now looking again at it..)

![image](https://github.com/Bertverbeek4PS/bc2adls/assets/44637996/d383c19a-4ab4-4cb6-85da-178671c3760a)
![image](https://github.com/Bertverbeek4PS/bc2adls/assets/44637996/89575e02-16a7-4e9e-b9ff-6189bb7acd60)

So at this moment we're using only blocks when the Storage Type is set to Azure Data Lake and makes this sentence below not valid I believe.
```Note that each upload creates a new block within the blob in the data lake. The size of such blocks are constrained as described at [Put Block (REST API) - Azure Storage | Microsoft Docs](https://docs.microsoft.com/en-us/rest/api/storageservices/put-block#remarks).```

I didn't test the combination of applying the method of Azure Data Lake (using blocks) with an endpoint of OneLake, because the documentation doesn't explicit mentions this.
```This page details the changes between ADLS Gen2 and OneLake when calling DFS APIs and other differences between the two services.```
https://learn.microsoft.com/en-us/fabric/onelake/onelake-api-parity

Does it make sense to verify again if the block approach could work with the OneLake API? Do you perhaps have any information on this?